File: C++ LAB/inharitance.cpp
#include<iostream>
using namespace std;
class Student{
    public:
    string name;
    int age ;
    void get(){
        cout<<"Student Name is : "<<name<<endl;
        cout<<"Student Age is : "<<age<<endl;
    }
};
class Human : public Student {
   public:
   int id;
   void print(){
    cout<<"Id is : "<<id<<endl;
    get();
   }
};
int main ()
{
Human h1;
h1.name = "Abdullah";
h1.age = 15;
h1.id = 10;
h1.print();
return 0;
}

File: Python/inharitance.py
class base(object) :
    def sign_in(self):
        print("Sign in done")
        return "done"
class Derived(base):
    def __init__(self,name, age ):
        self.name = name
        self.age = age

    def attack(self):
        print(f"Derived class Name is {self.name} Age is {self.age} ")

class expert(Derived):
    def __init__(self,name,age):
        self.name = name
        self.age = age

    def attack(self):
       print(f"Expert class Name is {self.name} Age is {self.age} ")

base1 = base()
print(base1.sign_in())
expert1 = expert("Abdullah",10)
print(expert1.sign_in())
Derived1 = Derived("Imran",20)
print(Derived1.sign_in())
print(isinstance(expert1,Derived))
# Polymorphism in python 
def f(ch):
    ch.attack()
f(Derived1)
f(expert1)

for item in [Derived1,expert1]:
    item.attack()